pkg/analyzer/config.go

```diff
@@ -15,18 +15,18 @@ type ParsedNSXConfig interface {
 	AnalyzedConnectivity() connectivity.ConnMap
 	DFW() *dfw.DFW
 	DefaultDenyRule() *dfw.FwRule
-	VMs() []*endpoints.VM
-	VMToGroupsMap() map[*endpoints.VM][]*collector.Group
+	VMs() []endpoints.EP
+	VMToGroupsMap() map[endpoints.EP][]*collector.Group
 	GetGroups() []*collector.Group
 }
 
 // config captures nsx config, implements NSXConfig interface
 type config struct {
-	vms         []*endpoints.VM                      // list of all vms
-	vmsMap      map[string]*endpoints.VM             // map from uid to vm objects
-	Fw          *dfw.DFW                             // currently assuming one DFW only (todo: rename pkg dfw)
-	Groups      []*collector.Group                   // list of all groups (also these with no vms)
-	GroupsPerVM map[*endpoints.VM][]*collector.Group // map from vm to its groups
+	vms         []endpoints.EP                      // list of all vms
```

Comment:
cannot keep concrete VM type where relevant?
and use the `ep` interface for relevant cases (such as `connMap`) ?

Reply:
there is downside for that.
for example: 
`	res.InitPairs(false, c.vms, vmsFilter)`
which initial the connectivity map, should get a  slice of EPs, 
if the vms will be a slice of VMs, there should be a conversion, which I did not implement yet.



For now, I didnt distinguished between the cases and change to EP everywhere.
when I will dive to implementation, I will move back to *VM in places there are vms.